### DISCUSSION TITLE
Most Consistent and Profitable Setup using Screeni-py

### DISCUSSION THREAD

**Discussion by pranjal-joshi** [pranjal-joshi/Screeni-py#95] [Show and tell]
## Welcome to Screeni-py Discussion! 😄 
  
  We’re using Discussions as a place to connect with other traders of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas about different setups and configurations.
  * Engage with other community members.

## Share your Setups: 📈 
  We would like to know if you have found some awesome configuration combination that helps you to get consistent profits from the markets! :money_mouth_face:
Please share such setup configurations here. We will keep you posted about it! :1st_place_medal:

### Note: ❗ 
Please do NOT post any bug/issue/feature request here. Please open an independent issue [here](https://github.com/pranjal-joshi/Screeni-py/issues) for such discussions.

**Comment by pranjal-joshi** [comment]
## Setup - The Momentum Gainers

### Setup Config
1. Select Index: All stocks
2. Select Criteria: 6 > Screen for reversal >  then Option 3 > Screen for Momentum Gainers (Rising Bullish Momentum)

### Chart Examples
1. Observe the 3 Candles, Spread (Body Size) of each candle is increasing with time.
2. Volume for each candle is consistently rising.
3. All three candles are either Gap-ups or at least `Open(t) >= Close(t-1)`
4. Prefer **Momentum Gainer** pattern forming nearby moving averages and candles are relatively small. Avoid trading on extended stocks where the price is way too far from MAs or the chart is showing a steep upward move like 90°

![MomentumGainer](https://user-images.githubusercontent.com/6128978/149926898-c27465ea-5e49-4e6a-b508-81007659c204.png)

### How to Trade?
Look at the result charts where `Momentum Gainer` is shown in the pattern parameter.
If the Stock is forming small candles with increasing spread and still away from its resistance level, then there would be a high probability of profit 💯

**Reply by dopevog** [reply]
@pranjal-joshi Hi, I had a few questions about this setup:

1. How far can the stock price be from the moving average
2. Is it fine if the spread is not increasing in all the candles?

Thanks!

**Reply by pranjal-joshi** [reply]
> @pranjal-joshi Hi, I had a few questions about this setup:
> 
> 1. How far can the stock price be from the moving average
> 2. Is it fine if the spread is not increasing in all the candles?
> 
> Thanks!

Hi @dopevog 

1. Stock prices can be considered as independent from MAs. But prefer if it's already in an uptrend

2. A spread of candles can be ignored. But every candle should be gap-up of previous and also take attention to the volumes.

Apart from this, take a look at the chart manually to check various timeframes, trendlines, support and resistance zones and only then take a trade if you think it's a favourable setup.

**Reply by dopevog** [reply]
@pranjal-joshi please could you emphasise how you use this pattern to select stocks?

**Reply by pranjal-joshi** [reply]
Hi @dopevog 

As asked in #192 I manually go through all momentum gainers once a day and take a look at the chart manually.
I take a bullish swing trade where I found the pattern is occurring while the overall spread of the candles is still less than 2.5-3%

If you get momentum gainer for large candles, it seems like the rally has already happened and you can't confidently expect more in the short term.

Here is one more example that I just entered yesterday which was identified as a `momentum gainer`

![image](https://github.com/pranjal-joshi/Screeni-py/assets/6128978/bf0f8177-a023-4f32-9afe-a0396d61d038)

**Reply by dopevog** [reply]
Thank you a lot @pranjal-joshi!

**Reply by dopevog** [reply]
Hey @pranjal-joshi could you share some more setups which you would/did trade using the momentum gainer pattern so we can get more clarity on what to look for, when possible?

**Reply by pranjal-joshi** [reply]
The screenshot attached as a chart Example in the first post was of FSL.
You will find these patterns in almost every stock/index if you just watch historical charts. I don't think there is anything more that needs to be explained explicitly.
The key is to find the best R:R with multiple confirmations, plan SL properly, and pay attention to the chart instead of blindly jumping into "any" momentum gainer! @dopevog

**Comment by pranjal-joshi** [comment]
## Setup - IPO Base Breakout

### Setup Config
1. Select Ticker Group / Index: 13 > Newly Listed
2. Select Criteria: 0 / 2 / 7 - Full Screen / Breakout with Volume / Inside bar for last N days

### Chart Examples
1. Huge upside momentum is possible when the stock crosses its listing price and creates ATH (All-time High) for the first time.
2. There could be several types of IPO base breakouts, a few of them which occur most frequently are shown here as an example.

| Rounding Bottom| Inside Bar/Trendline | Both |
| --- | --- | --- |
| ![IPO1](https://user-images.githubusercontent.com/6128978/149931060-0c40bae4-aec7-4e8d-9376-2a6fb8febbe6.png) | ![IPO2](https://user-images.githubusercontent.com/6128978/149931109-7d447172-5185-4ac8-bac5-5313e5113641.png) | ![IPO3](https://user-images.githubusercontent.com/6128978/149931150-bb29362e-248b-479d-a2cf-0c48b3a3cc7e.png) |

### How to Trade?
1. Observe and Record the Breakout levels.
2. Set alerts for Breakout price.
3. Pay extra attention to volumes at which the price moves.
4. You may place Buy SL orders so that you won't (usually) miss the breakout when it breaks the resistance level.

**Comment by sunel** [comment]
# Breakout + Breakout Soon Stock Picks!

### Setup Config

1. Select Index: All stocks
2. Select Criteria: 2 > Screen for the stocks with recent Breakout & Volume

### Chart Examples
| 1 | 2 | 3 |
| --- | --- | --- |
| ![image](https://user-images.githubusercontent.com/1009777/150647011-3a41da68-3e3d-49e7-875d-256dfb6b2055.png) | ![image](https://user-images.githubusercontent.com/1009777/150647042-2d80ff53-6098-4cb1-8bbe-feb5ff8cb159.png) | ![image](https://user-images.githubusercontent.com/1009777/150647053-74da3965-0c1d-41eb-8248-d8e02c1b8829.png) |

### screenipy.ini

```
[config]
period = 300d
daystolookback = 30
duration = 1d
minprice = 3.0
maxprice = 2500
volumeratio = 2
consolidationpercentage = 10
shuffle = y
cachestockdata = y
onlystagetwostocks = y
useema = n
```

**Reply by pranjal-joshi** [reply]
Great Setup @sunel!

Can you share you `screenipy.ini` file that was used while this screening here for further clarification?
This file contains some important parameters like past how many days the software is observing to calculate the Break-out levels.

**Reply by sunel** [reply]
added

**Comment by pranjal-joshi** [comment]
## Setup - Narrow Range Breakout/Reversal (NR-4)

### Wait! But what is a Narrow Range? 
Click [Here](https://www.elearnmarkets.com/blog/nr4-and-nr7-trading-strategy-setup/) to read the basics about NR.

### Setup Config
1. Select Ticker Group / Index: Any
2. Select Criteria: Default / 6 / 6 - Full Screening / Reversal / Narrow Range
3. Enter NR timeframe = 4 (Days)

### Chart Examples
1. Day trading possible in high volatile markets (Ensure proper position sizing and stop-loss)
2. Trend continuation after a brief half and breaking out from the narrowed range for further move.

| Trend Reversal | Trend Continuation |
| --- | --- |
| ![NR4-Reversal](https://user-images.githubusercontent.com/6128978/153903324-353eac7f-c31e-4c6f-8ebb-32d8488a9439.png) | ![NR4-Breakdown](https://user-images.githubusercontent.com/6128978/153903392-14d250b1-abd2-4043-bf1a-f02410332ef3.png) |

### How to Trade?
1. Observe the candle range and volumes. Pay attention to candle details like color, candle spread, size of wicks, etc.
2. Observe Gap-Up/Down opening in the next session.
3. Usually needs a small SL as the market is in a narrow range, so high probability of good Risk:Reward ratio.

### Note:
The Narrow Range screening is a dynamic algorithm. It works differently depending on the time of screening.
1. During Live-market hours, It screens for stocks formed narrow range earlier and opened Gap-Up/Down today so you can get an Intraday trade.
2. During After-market hours, It screens for the stocks which have formed a narrow range today and could possibly break-out in upcoming trading sessions.

### Configuration used:

The following configuration `(screenipy.ini)` was used while screening for NR4.
```
[config]
period = 300d
daystolookback = 30
duration = 1d
minprice = 30.0
maxprice = 10000
volumeratio = 2
consolidationpercentage = 10
shuffle = y
cachestockdata = y
onlystagetwostocks = y
useema = n
```

**Comment by pranjal-joshi** [comment]
## Setup - Buy at Trendline for Swing/Mid/Long Term

### Setup Config
1. Select Ticker Group / Index: Any
2. Select Criteria: Default / 7 / 5 - Full Screening / Chart Patterns / Buy at Trendline
3. Adjust `period = N` from `screenipy.ini` as per your timeframe to draw the trendline from last N days

### Chart Example

| Trend Continuation |
| --- |
| ![image](https://user-images.githubusercontent.com/6128978/183817139-9c059d92-6826-4e3c-96f1-f3b92343694a.png) |

### How to Trade?
1. Ideal for buying a trending stock near its support trendline in longer timeframes.
2. Buying near Trendline keeps the SL small and offers a high probability trade!

### Configuration used:

The following configuration `(screenipy.ini)` was used while screening for Buy at Trendline.
```
[config]
period = 400d
daystolookback = 30
duration = 1d
minprice = 30.0
maxprice = 15000
volumeratio = 2
consolidationpercentage = 10
shuffle = y
cachestockdata = y
onlystagetwostocks = y
useema = n
```

**Comment by anirbanphys** [comment]
Dear @pranjal-joshi,
As you suggested before, I did a back testing and got good results for IMPAL.NS. 
I changed the configuration as suggested.
One doubt, today the market is bearish, the utility also predicted that perfectly yesterday. If you see, few stocks perform very well while the market goes down for a particular day, the very next day few other stcoks replace that spots. How screenipy.ini configuration can detect this criteria for the very next day?

Thanks,
Anirban

**Reply by pranjal-joshi** [reply]
Sorry for the delayed reply.
I was unable to understand your query. Can you explain further? @anirbanphys

**Comment by anirbanphys** [comment]
Dear @pranjal-joshi, 
Very sorry for the late response.
Let me try to explain my doubts. On a day when the market is bearish, a few stocks perform very well and provide good intraday returns. I meant the day's "gainers." Is there any way to filter those stocks?

**Comment by pranjal-joshi** [comment]
## Setup - Buy signal from Lorentzian Classification Indicator

### Setup Config
1. Select Ticker Group / Index: Any
2. Select Criteria: 6 > 7 - Reversal signals / Lorentzian Classification
3. Select Buy/Sell/Any Signal as per your bullish/bearish view of the market. Keep Any for both signals.

### Chart Example

| Lorentzian Classification - Chart Reading Example |
| --- |
| ![image](https://github.com/pranjal-joshi/Screeni-py/assets/6128978/0a3fd413-07de-467a-8cc0-62cfad094145) |
| Green signals below the candles are Buy signals and Red signals above the candles are Sell signals |

### How to Trade?
1. Lorentzian Classification is a Machine Learning (AI) based indicator (Invented by @jdehorty and available freely on TradingView) to find the Buy/Sell signals as it analyses multiple Indicators like RSI, CCI, WT, ADX together to determine if the Buy/Sell events are generated.
2. Execute the trade when Candle is closed (or about to close) and Buy/Sell signal is generated
3. Use this indicator in confluence with other indicators or trendline breaks for high probability trades

### Configuration used:

The following configuration `(screenipy.ini)` was used while screening for Lorentzian Classification but author suggest that it work on any timeframe as it doesn't have temporal dependencies in it's input features.

```
[config]
period = 300d
daystolookback = 30
duration = 1d
minprice = 30.0
maxprice = 15000
volumeratio = 2
consolidationpercentage = 10
shuffle = y
cachestockdata = y
onlystagetwostocks = y
useema = n
```
